ui/panels-plugin/src/plugins/time-series-chart/TimeSeriesChartPanel.test.tsx

```diff
@@ -115,7 +127,92 @@ describe('TimeSeriesChartPanel', () => {
 
   it('should toggle selected state when a legend item is clicked', async () => {
     renderPanel();
-    await userEvent.click(screen.getByTestId('legend-item-hsla(-141599372,50%,50%,0.8)'));
-    expect(screen.getByTestId('legend-item-hsla(-141599372,50%,50%,0.8)')).toHaveClass('Mui-selected');
+
+    const seriesArr = Array.from(MOCK_TIME_SERIES_DATA.series);
+    const firstLegend = getLegendByName(seriesArr[0]?.name);
+    const secondLegend = getLegendByName(seriesArr[1]?.name);
+
+    userEvent.click(firstLegend);
+    expect(firstLegend).toHaveClass('Mui-selected');
+    expect(secondLegend).not.toHaveClass('Mui-selected');
```

Comment:
From an accessibility perspective, we may want to adjust the HTML for the legend to be able to communicate to a screen reader that these are selected, but I don't want to deal with fixing that in this particular PR.